proptest/src/sugar.rs

```diff
@@ -1206,6 +1257,14 @@ mod test {
         }
     }
 
+    proptest! {
+        #[tokio::test]
+        async fn test_something_async(a in 0u32..42u32, b in 1u32..10u32) {
+            prop_assume!(a != 41 || b != 9);
```

Comment:
If you add an `await` call into the body this will fail to compile, even something like `async { }.await`.

Reply:
From expanding the macro it looks like this is not actually adding the `async` keyword in front of the function, but I'm not sure why. Annoying that I didn't spot this before.

Reply:
Yeah I spent a bit of time trying to figure out how to fix, but I'm not familiar with that code so couldn't puzzle it out.  _seems_ like it should be straightforward, but macros are always harder than you think they'll be :)